### PR TITLE
Add caching support to the tenancy client

### PR DESCRIPTION
**Pull request by jongeorge1**
Adds support to the tenancy client library for honouring caching headers returned from the service.

### CODE REVIEW

File: Solutions/Marain.Tenancy.Specs/Integration/Features/TenancyClientWithCachingEnabled.feature [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	When I get the tenant with the id called "ChildTenantId" and the ETag called "ChildTenantETag"
+	Then it should throw a TenantNotModifiedException
+
+Scenario: Get a tenant with an etag retrieved from a tenant got from the repo
+	Given I create a child tenant called "ChildTenant1" for the root tenant
+	And I get the tenant id of the tenant called "ChildTenant1" and call it "ChildTenantId"
+	And I get the tenant with the id called "ChildTenantId" and call it "Result"
+	And I get the ETag of the tenant called "Result" and call it "ResultETag"
+	When I get the tenant with the id called "ChildTenantId" and the ETag called "ResultETag"
+	Then it should throw a TenantNotModifiedException
```
Comment (idg10) [inline Solutions/Marain.Tenancy.Specs/Integration/Features/TenancyClientWithCachingEnabled.feature:36]: Isn't "tenant not modified" the normal case? Throwing an exception seems odd here.

That said, I'm guessing this behaviour wasn't actually a change introduced in this PR?

File: Solutions/Marain.Tenancy.Specs/Marain.Tenancy.Specs.csproj [hunk 2]
```diff
@@ -16,6 +16,10 @@
     <NoWarn>SA0001;SA1204;SA1600;SA1602;CS1591</NoWarn>
   </PropertyGroup>
 
+  <ItemGroup>
+    <SpecFlowObsoleteCodeBehindFiles Remove="Integration\Features\TenancyClientWithCachingDisabled - Copy.feature.cs" />
```
Comment (idg10) [inline Solutions/Marain.Tenancy.Specs/Marain.Tenancy.Specs.csproj:20]: I suspect you don't want this.

File: Solutions/Marain.Tenancy.Specs/Marain.Tenancy.Specs.csproj [hunk 3]
```diff
@@ -29,6 +33,12 @@
     <ProjectReference Include="..\Marain.Tenancy.OpenApi.Service\Marain.Tenancy.OpenApi.Service.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Compile Update="Integration\Features\TenancyClientWithCachingEnabled.feature.cs">
```
Comment (idg10) [inline Solutions/Marain.Tenancy.Specs/Marain.Tenancy.Specs.csproj:37]: Do we need this? I thought these relationships were supposed to be detected automatically by SpecFlow now.

File: Solutions/Marain.Tenancy.Specs/Marain.Tenancy.Specs.csproj [hunk 4]
```diff
@@ -37,4 +47,11 @@
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </None>
   </ItemGroup>
+
+  <ItemGroup>
+    <SpecFlowFeatureFiles Update="Integration\Features\TenancyClientWithCachingEnabled.feature">
+      <Visible>$(UsingMicrosoftNETSdk)</Visible>
+      <CodeBehindFile>%(RelativeDir)%(Filename).feature$(DefaultLanguageSourceExtension)</CodeBehindFile>
+    </SpecFlowFeatureFiles>
+  </ItemGroup>
```
Comment (idg10) [inline Solutions/Marain.Tenancy.Specs/Marain.Tenancy.Specs.csproj:56]: This all looks odd.

File: Solutions/Marain.Tenancy.Specs/Integration/Features/TenancyClientWithCachingEnabled.feature (hunk 1 above)
Comment (jongeorge1) [reply Solutions/Marain.Tenancy.Specs/Integration/Features/TenancyClientWithCachingEnabled.feature:36]: Nope, that's been there for ages.

**Review by idg10 — APPROVED** [review]

**Pull request merged**